src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -179,26 +179,24 @@
   (rr/get-header request "If-None-Match"))
 
 (schema/defn ^:always-validate
-  basic-manifest-info-from-jruby->basic-manifest-info-for-json
-    :- {schema/Any schema/Any}
+  add-path-to-file-entry :- Map
   "Convert the value for a manifest file entry into an appropriate map for
   use in serializing an environment_classes response to JSON."
-  [file-info :- Map$Entry]
-  (let [file-detail (val file-info)]
-    (if (instance? Map file-detail)
-      (into {} file-detail)
-      {:classes file-detail})))
+  [file-detail :- Map
+   file-name :- schema/Str]
+  (.put file-detail "path" file-name)
```

Comment:
@cprice404 This is a follow-up to the optimization idea that we talked about in https://github.com/camlow325/puppet-server/commit/f797e3bd664a41f910afee2126d8fa5791971390#commitcomment-15958208.  I'm okay with going back to constructing a Clojure map and `assoc`ing the `:path` key/value pair to it - more like what we had before - if this seems too presumptuous.  WDYT?